mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -695,6 +699,22 @@ public Void execute() throws MailboxException {
         return results;
     }
 
+    private boolean containChildren (final Mailbox parentMailbox, final List<Mailbox> mailboxesWithPathLike, final MailboxSession mailboxSession) {
+        return FluentIterable.from(mailboxesWithPathLike)
+            .anyMatch(isChildren(parentMailbox, mailboxSession));
+    }
+
+    private Predicate<Mailbox> isChildren(final Mailbox parentMailbox, final MailboxSession mailboxSession) {
```

Comment:
I know there are multiple implementations of Mailbox; but it could be great to have a method like this:
Mailbox.isChildOf(mailbox)